language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java

```diff
@@ -1222,4 +1223,23 @@ public static Boolean isInFunctionCallParameterContext(PositionedOperationContex
                 && (!node.closeParenToken().isMissing())
                 && (cursorPosition <= node.closeParenToken().textRange().startOffset());
     }
+
+    /**
+     * Check if the cursor is positioned in a lock statement node context.
+     *
+     * @param context Completion context.
+     * @return {@link Boolean} whether the cursor is in lock statement node context.
+     */
+    public static Boolean withinLockStatementNode(BallerinaCompletionContext context) {
+        NonTerminalNode evalNode = context.getNodeAtCursor();
+        boolean withinLockStatementNode = false;
+        do {
+            if (evalNode.kind() == SyntaxKind.LOCK_STATEMENT) {
+                withinLockStatementNode = true;
+            }
+            evalNode = evalNode.parent();
+        }
+        while (!withinLockStatementNode && evalNode != null);
+        return withinLockStatementNode;
```

Comment:
We don't need the `withinLockStatementNode` right?
```suggestion
        boolean withinLockStatementNode = false;
        do {
                    if (evalNode.kind() == SyntaxKind.LOCK_STATEMENT) {
                            return true;
                    }
                    evalNode = evalNode.parent();
       } while (evalNode != null);
       return false;
```

Reply:
Agreed!